pennylane/measurements/measurements.py

```diff
@@ -693,3 +695,40 @@ def map_wires(self, wire_map: dict):
         else:
             new_measurement._wires = Wires([wire_map.get(wire, wire) for wire in self.wires])
         return new_measurement
+
+
+class SampleMeasurement(MeasurementProcess, ABC):
+    """Sample-based measurement process."""
+
+    @abstractmethod
+    def process(
+        self, samples: Sequence[complex], shot_range: Tuple[int] = None, bin_size: int = None
```

Comment:
This is up for debate, but IMO I think it would make more sense for this method to take a `device` argument instead of what's given here. One advantage for this is that the signature for this function is the same across all measurement processes. My other comment is also relevant here.

Reply:
But as I mentioned last week, the `classical_shadow` measurement is a special case. Honestly I believe it should be considered as a tape transforms instead of a measurement: the tape transform would generate `n_shots` tapes with `shots=1`, execute them on the device, and then generate the outcomes and recipes in the classical post processing.

Reply:
> To break the coupling between measurements and devices we need to define a set of arguments/variables that are sufficient for computing any measurement process.

This is a similar approach to the one followed in the new device API refactor: they want to introduce a class `ExecutionConfig` that will contain all the needed information for the execution of a tape.

Reply:
I agree with you that right now it would make more sense for the device to handle all the shot information. However in the new device API we are planning on moving shots into an "`ExecutionConfig`" class that will be passed around. That is why I decided to also pass the shots information into the measurement class, to comply with the new refactor.